### PR TITLE
1822: Upgrade CI docker image to Alpine 3.16

### LINKED ISSUE #1822: Upgrade CI docker image to Alpine 3.16
**What Needs to be Done?**

Alpine Linux version 3.16 was just released. Our current docker image built on Alpine uses 3.13. We should upgrade, so that our testing is reflective of what people are likely to deploy.

### PR DESCRIPTION
**Pull request by cz4rs**
fixes #1822

### CODE REVIEW

**Comment by jstrzebonski** [comment]
> @jstrzebonski Can you please update the YAML file path for alpine build to `ci/azure/azure-clang-alpine-mpich.yml`?

Done.

**Comment by jstrzebonski** [comment]
Oh, and I've also renamed the pipeline to `PR tests (alpine, clang-13, mpich)` (as is in yml file).

File: ci/azure/azure-clang-alpine-mpich.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ############## Warning this is a generated file---do not modify ###############
 ###############################################################################
 
-name: PR tests (clang-10, alpine, mpich)
+name: PR tests (alpine, clang-13, mpich)
```
Comment (nlslatt) [inline ci/azure/azure-clang-alpine-mpich.yml:5]: Why switch the order? It seems like the rest list the compiler first.

**Review by nlslatt — COMMENTED** [review]
The changes look reasonable to me, but I'm not sure what motivated one change.

**Review by nlslatt — APPROVED** [review]
For consistency, I think we should have the compiler named first, but let's get this merged in so we can unblock PR testing.

**Pull request merged**